src/vector.h

```diff
@@ -0,0 +1,46 @@
+#ifndef VECTOR_H
+#define VECTOR_H
+
+#include "log.h"
+
+#include <stdint.h>
+#include <stdbool.h>
+
+typedef struct vector {
```

Comment:
We can do this with an opaque interface.  Example:

https://github.com/lyft/statsite/blob/master/src/strbuf.h
https://github.com/lyft/statsite/blob/master/src/strbuf.c